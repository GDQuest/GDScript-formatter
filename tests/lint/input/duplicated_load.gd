#region Good
const GOOD_PRELOAD_1 = preload("res://some/path1.tscn")
const GOOD_PRELOAD_2 = preload("res://some/path2.tscn")

var good_load_1 = load("res://another/path1.tscn")
var good_load_2 = load("res://another/path2.tscn")

var good_scene_path = "res://some/path3.tscn"
var good_load_3 = load(good_scene_path)
var good_load_4 = load(good_scene_path)

#endregion


#region Bad
const BAD_PRELOAD_1 = preload("res://duplicate/path.tscn")
const BAD_PRELOAD_2 = preload("res://duplicate/path.tscn")

var bad_load_1 = load("res://another/duplicate/path.tscn")
var bad_load_2 = load("res://another/duplicate/path.tscn")

#endregion