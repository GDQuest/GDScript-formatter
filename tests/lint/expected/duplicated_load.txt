tests/lint/input/duplicated_load.gd:16:duplicated-load:warning: Duplicated load of '"res://duplicate/path.tscn"'. Consider extracting to a constant.
tests/lint/input/duplicated_load.gd:17:duplicated-load:warning: Duplicated load of '"res://duplicate/path.tscn"'. Consider extracting to a constant.
tests/lint/input/duplicated_load.gd:19:duplicated-load:warning: Duplicated load of '"res://another/duplicate/path.tscn"'. Consider extracting to a constant.
tests/lint/input/duplicated_load.gd:20:duplicated-load:warning: Duplicated load of '"res://another/duplicate/path.tscn"'. Consider extracting to a constant.